### ISSUE TITLE
Error Building for ARM

### ISSUE THREAD

**Issue by arranf** [psychon/x11rb#297]
`cross build --target arm-unknown-linux-gnueabihf --release` in a project using your library produces the following output.

```
error: failed to run custom build command for `x11rb v0.4.1`

Caused by:
  process didn't exit successfully: `/target/release/build/x11rb-b1c49fc1ef8ff99d/build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-changed=rs_code_generator.py
cargo:rerun-if-changed=code_generator_helpers/special_cases.py
cargo:rerun-if-changed=code_generator_helpers/__init__.py
cargo:rerun-if-changed=code_generator_helpers/request.py
cargo:rerun-if-changed=code_generator_helpers/module.py
cargo:rerun-if-changed=code_generator_helpers/output.py
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/xcbgen/error.py
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/xcbgen/xtypes.py
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/xcbgen/state.py
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/xcbgen/align.py
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/xcbgen/expr.py
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/xcbgen/__init__.py
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/xcbgen/matcher.py
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/randr.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xv.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xproto.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xtest.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xf86vidmode.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/dri3.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xkb.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/record.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xf86dri.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/res.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/composite.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xc_misc.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/dpms.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/present.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xvmc.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xinput.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/sync.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/shm.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/bigreq.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xprint.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/ge.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/damage.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/dri2.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/glx.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/shape.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xfixes.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xinerama.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xevie.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/render.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/xselinux.xml
cargo:rerun-if-changed=xcbproto-1.13-6-ge79f6b0/src/screensaver.xml

--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /cargo/registry/src/github.com-1ecc6299db9ec823/x11rb-0.4.1/build.rs:61:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by psychon** [comment]
`build.rs:61:18` is the call to the python code generator. And apparently there is no error message from it. Uhm... do you have python on that system? If starting `python` fails, that would explain the `NotFound` error.
If this really is a missing Python installation, then we should improve that error message, somehow.

Alternatively, could you run the following commands in the directory `~/.cargo/registry/src/github.com-*/x11rb-0.4.1` and check how it fails?
```
mkdir /tmp/output-for-x11rb
python rs_code_generator.py -p xcbproto-1.13-6-ge79f6b0 -i xcbproto-1.13-6-ge79f6b0/src -o /tmp/output-for-x11rb mod
```

**Comment by arranf** [comment]
I'm pretty confident it's not a lack of python. Both `python` and `python3` work.
That works absolutely fine!

I guess the issue is I'm running it with `cross` which _may_ not have `python` installed in its build image. I'll do a little more digging.

**Comment by psychon** [comment]
Oh, by the way: If you can, you could try building the git version. Instead of running the code generator at build time, the generated code is now part of the git repo. That could make your debug-`build.rs` work unnecessary.

**Comment by eduardosm** [comment]
@arranf Could you please try the recently-released 0.5 version? This new version does not have a `build.rs` script.
